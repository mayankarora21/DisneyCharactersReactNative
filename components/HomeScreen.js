import { useEffect } from 'react';
import {View, FlatList, StyleSheet} from 'react-native';
import SearchBar  from './SearchBar';
import CharacterItem from './CharacterItem';

import { getCharacterList } from '../store/character-actions';
import {useSelector, useDispatch} from 'react-redux';

const HomeScreen = () => {
    const dispatch = useDispatch();
    const filteredDisneyCharacterList = useSelector(state => state.filteredDisneyCharacterList);
    
    useEffect(() => {
        dispatch(getCharacterList());
    }, [])


    return (
        <View style = {styles.container}>
            <SearchBar></SearchBar>
            <FlatList data = {filteredDisneyCharacterList} keyExtractor = {(item, index) => index} renderItem = {(itemData) => {
                return (
                    <CharacterItem character = {itemData.item}></CharacterItem>
                );
            }}></FlatList>
        </View>
    );
}

export default HomeScreen;


const styles = StyleSheet.create({
    container: {
        padding: 15,
        flex: 1,
    }
})