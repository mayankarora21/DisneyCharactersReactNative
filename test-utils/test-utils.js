import React from 'react'
import {render} from '@testing-library/react-native';
import store from '../store/store';
import { Provider } from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

const AllTheProviders = ({children}) => {
  return (
    <Provider store = {store}>
      <NavigationContainer>
            {children}
      </NavigationContainer>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}