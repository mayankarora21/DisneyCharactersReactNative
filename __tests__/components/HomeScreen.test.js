import {fireEvent, render, screen} from '../../test-utils/test-utils';
import HomeScreen from '../../components/HomeScreen';
import axios from 'axios';

jest.mock('axios');
const characterListResponse = {
    data: {
        data: [
            {name: "Character 1", imageUrl: "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png"},
            {name: "Character 2", imageUrl: "https://static.wikia.nocookie.net/disney/images/3/3f/90%27s_Adventure_Bear_profile.png"},
            {name: "Character 3", imageUrl: "https://static.wikia.nocookie.net/disney/images/1/1e/Profile_-_Ace.png"},
        ]
    }
}

axios.get.mockResolvedValue(characterListResponse);

test('HomeScreen should render correctly', async () => {
    render(<HomeScreen/>);
    await screen.findByText('Character 1');
    expect(screen.toJSON()).toMatchSnapshot();
})


test('character list should be filtered according to search text', async () => {
    render(<HomeScreen/>);
    await screen.findByText('Character 1');
    expect(screen.queryByText('Character 2')).toBeTruthy();
    expect(screen.queryByText('Character 3')).toBeTruthy();
    
    fireEvent.changeText(screen.getByPlaceholderText('Search'), '1');
    await screen.findByText('Character 1');
    expect(screen.queryByText('Character 2')).toBeFalsy();
    expect(screen.queryByText('Character 3')).toBeFalsy();
})