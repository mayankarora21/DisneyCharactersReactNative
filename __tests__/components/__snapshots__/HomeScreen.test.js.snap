// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View>
    <TextInput
      onChangeText={[Function]}
      placeholder="Search"
      style={
        {
          "borderWidth": 1,
          "padding": 10,
        }
      }
    />
  </View>
  <RCTScrollView
    data={
      [
        {
          "imageUrl": "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png",
          "name": "Character 1",
        },
        {
          "imageUrl": "https://static.wikia.nocookie.net/disney/images/3/3f/90%27s_Adventure_Bear_profile.png",
          "name": "Character 2",
        },
        {
          "imageUrl": "https://static.wikia.nocookie.net/disney/images/1/1e/Profile_-_Ace.png",
          "name": "Character 3",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text>
            Character 1
          </Text>
          <Image
            source={
              {
                "uri": "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png",
              }
            }
            style={
              {
                "borderRadius": 50,
                "height": 100,
                "resizeMode": "stretch",
                "width": 100,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text>
            Character 2
          </Text>
          <Image
            source={
              {
                "uri": "https://static.wikia.nocookie.net/disney/images/3/3f/90%27s_Adventure_Bear_profile.png",
              }
            }
            style={
              {
                "borderRadius": 50,
                "height": 100,
                "resizeMode": "stretch",
                "width": 100,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text>
            Character 3
          </Text>
          <Image
            source={
              {
                "uri": "https://static.wikia.nocookie.net/disney/images/1/1e/Profile_-_Ace.png",
              }
            }
            style={
              {
                "borderRadius": 50,
                "height": 100,
                "resizeMode": "stretch",
                "width": 100,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
